src/mkvgen/src/NalAdapter.c

```diff
@@ -132,7 +138,9 @@ STATUS adaptFrameNalsFromAnnexBToAvcc(PBYTE pFrameData,
     if (STATUS_SUCCEEDED(retStatus) && pAdaptedFrameDataSize != NULL) {
         // NOTE: Due to EPB removal we could in fact make the adaptation buffer smaller than the original
         // We will require at least the original size buffer.
-        *pAdaptedFrameDataSize = MAX(frameDataSize, (UINT32)(pAdaptedCurPnt - pAdaptedFrameData));
+        *pAdaptedFrameDataSize = (removeEpb && HANDLING_TRAILING_NALU_ZERO) ?
```

Comment:
shouldnt pAdaptedCurPnt - pAdaptedFrameData also account for the frameDataSize?

Reply:
In case of no EPB (which is the case where we need to split the runs for CPD for example instead of the frame NALs) the size will be smaller with the broken encoder hack. Hence, we can't use the original frame size. The adapted frame size can be smaller in this case.

Reply:
In case of no EPB (which is the case where we need to split the runs for CPD for example instead of the frame NALs) the size will be smaller with the broken encoder hack. Hence, we can't use the original frame size. The adapted frame size can be smaller in this case.